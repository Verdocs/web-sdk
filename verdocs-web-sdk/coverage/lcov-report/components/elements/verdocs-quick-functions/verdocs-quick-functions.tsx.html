
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components/elements/verdocs-quick-functions/verdocs-quick-functions.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">components/elements/verdocs-quick-functions</a> verdocs-quick-functions.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">63.63% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>7/11</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">33.33% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/6</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">63.63% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>7/11</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {VerdocsEndpoint} from '@verdocs/js-sdk';
import {Component, h, Event, EventEmitter, Prop} from '@stencil/core';
&nbsp;
const DocumentIcon = `&lt;svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="#000000"&gt;&lt;path d="M0 0h24v24H0V0z" fill="none"/&gt;&lt;path d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11z"/&gt;&lt;/svg&gt;`;
&nbsp;
const TemplateIcon = `&lt;svg xmlns="http://www.w3.org/2000/svg" enable-background="new 0 0 24 24" height="24px" viewBox="0 0 24 24" width="24px" fill="#000000"&gt;&lt;g&gt;&lt;rect fill="none" height="24" width="24"/&gt;&lt;g&gt;&lt;path d="M19,5v14H5V5H19 M19,3H5C3.9,3,3,3.9,3,5v14c0,1.1,0.9,2,2,2h14c1.1,0,2-0.9,2-2V5C21,3.9,20.1,3,19,3L19,3z"/&gt;&lt;/g&gt;&lt;path d="M14,17H7v-2h7V17z M17,13H7v-2h10V13z M17,9H7V7h10V9z"/&gt;&lt;/g&gt;&lt;/svg&gt;`;
&nbsp;
/**
 * Display quick-function buttons for creating templates and documents.
 *
 * Authentication is required to demonstrate this Element. You may do this in Storybook by using the Auth
 * embed. This Element will reuse the same session produced by logging in via that Embed.
 */
@Component({
  tag: 'verdocs-quick-functions',
  styleUrl: 'verdocs-quick-functions.scss',
})
export class VerdocsQuickFunctions {
  /**
   * The endpoint to use to communicate with Verdocs. If not set, the default endpoint will be used.
   */
  @Prop() endpoint: VerdocsEndpoint = VerdocsEndpoint.getDefault();
&nbsp;
  /**
   * Event fired when an entry is clicked.
   */
  @Event({composed: true}) createTemplate: EventEmitter;
&nbsp;
  /**
   * Event fired when an entry is clicked.
   */
  @Event({composed: true}) createDocument: EventEmitter;
&nbsp;
<span class="fstat-no" title="function not covered" >  handleCreateTemplate(</span>) {
<span class="cstat-no" title="statement not covered" >    this.createTemplate.emit();</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  handleCreateDocument(</span>) {
<span class="cstat-no" title="statement not covered" >    this.createDocument.emit();</span>
  }
&nbsp;
  render() {
    return (
      &lt;div class="container"&gt;
        &lt;p class="title"&gt;Quick Create&lt;/p&gt;
        &lt;button onClick={<span class="fstat-no" title="function not covered" >() =</span>&gt; <span class="cstat-no" title="statement not covered" >this.handleCreateTemplate()}</span> innerHTML={TemplateIcon}&gt;
          Template
        &lt;/button&gt;
        &lt;button onClick={<span class="fstat-no" title="function not covered" >() =</span>&gt; <span class="cstat-no" title="statement not covered" >this.handleCreateDocument()}</span> innerHTML={DocumentIcon}&gt;
          Document
        &lt;/button&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2025-07-24T13:19:16.585Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    