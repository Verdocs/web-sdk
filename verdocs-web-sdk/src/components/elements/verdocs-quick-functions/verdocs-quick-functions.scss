@import '../../../theme.scss';

verdocs-quick-functions {
  font-family: $verdocs-primary-font;

  .container {
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    width: 240px;
    background-color: $verdocs-bg-5;
    text-transform: uppercase;
    padding: 10px 10px 20px 10px;
    font-family: sans-serif;
    color: #ffffff;
  }

  .title {
    font-size: 22px;
    font-weight: 700;
    line-height: 30px;
    margin: 1em 0 40px 0;
  }

  button {
    width: 140px;
    height: 40px;
    border: none;
    display: flex;
    color: $verdocs-grey-4;
    cursor: pointer;
    font-size: 14px;
    font-weight: 700;
    line-height: 25px;
    align-items: center;
    margin-bottom: 20px;
    border-radius: 5px;
    text-transform: uppercase;
    justify-content: space-evenly;
    background-color: $verdocs-green;
  }

  button svg {
    fill: $verdocs-grey-4;
  }
}
