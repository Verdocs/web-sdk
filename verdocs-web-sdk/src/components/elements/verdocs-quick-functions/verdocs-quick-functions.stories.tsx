import {html} from 'lit-html';
import {Meta} from '@storybook/web-components';

export default {
  title: 'Elements/Search/Quick Functions',
  component: 'verdocs-quick-functions',
  args: {
    options: {
      title: 'Quick Create',
    },
  },
  argTypes: {
    createTemplate: {action: 'createTemplate'},
    createDocument: {action: 'createDocument'},
  },
} as Meta;

export const QuickFunctions = ({options, createTemplate, createDocument}) =>
  html`<verdocs-quick-functions .options=${options} @createTemplate=${createTemplate} @createDocument=${createDocument} tall />`;
