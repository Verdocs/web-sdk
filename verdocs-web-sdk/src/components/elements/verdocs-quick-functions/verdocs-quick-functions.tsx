import {VerdocsEndpoint} from '@verdocs/js-sdk';
import {Component, h, Event, EventEmitter, Prop} from '@stencil/core';

const DocumentIcon = `<svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="#000000"><path d="M0 0h24v24H0V0z" fill="none"/><path d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11z"/></svg>`;

const TemplateIcon = `<svg xmlns="http://www.w3.org/2000/svg" enable-background="new 0 0 24 24" height="24px" viewBox="0 0 24 24" width="24px" fill="#000000"><g><rect fill="none" height="24" width="24"/><g><path d="M19,5v14H5V5H19 M19,3H5C3.9,3,3,3.9,3,5v14c0,1.1,0.9,2,2,2h14c1.1,0,2-0.9,2-2V5C21,3.9,20.1,3,19,3L19,3z"/></g><path d="M14,17H7v-2h7V17z M17,13H7v-2h10V13z M17,9H7V7h10V9z"/></g></svg>`;

/**
 * Display quick-function buttons for creating templates and documents.
 *
 * Authentication is required to demonstrate this Element. You may do this in Storybook by using the Auth
 * embed. This Element will reuse the same session produced by logging in via that Embed.
 */
@Component({
  tag: 'verdocs-quick-functions',
  styleUrl: 'verdocs-quick-functions.scss',
})
export class VerdocsQuickFunctions {
  /**
   * The endpoint to use to communicate with Verdocs. If not set, the default endpoint will be used.
   */
  @Prop() endpoint: VerdocsEndpoint = VerdocsEndpoint.getDefault();

  /**
   * Event fired when an entry is clicked.
   */
  @Event({composed: true}) createTemplate: EventEmitter;

  /**
   * Event fired when an entry is clicked.
   */
  @Event({composed: true}) createDocument: EventEmitter;

  handleCreateTemplate() {
    this.createTemplate.emit();
  }

  handleCreateDocument() {
    this.createDocument.emit();
  }

  render() {
    return (
      <div class="container">
        <p class="title">Quick Create</p>
        <button onClick={() => this.handleCreateTemplate()} innerHTML={TemplateIcon}>
          Template
        </button>
        <button onClick={() => this.handleCreateDocument()} innerHTML={DocumentIcon}>
          Document
        </button>
      </div>
    );
  }
}
