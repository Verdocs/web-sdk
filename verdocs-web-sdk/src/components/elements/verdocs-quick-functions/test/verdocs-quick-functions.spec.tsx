import {newSpecPage} from '@stencil/core/testing';
import {VerdocsQuickFunctions} from '../verdocs-quick-functions';

describe('search-quick-functions', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [VerdocsQuickFunctions],
      html: `<verdocs-quick-functions></verdocs-quick-functions>`,
    });
    expect(page.root).toEqualHtml(`
      <verdocs-quick-functions>
        <mock:shadow-root>
          <slot></slot>
        </mock:shadow-root>
      </verdocs-quick-functions>
    `);
  });
});
