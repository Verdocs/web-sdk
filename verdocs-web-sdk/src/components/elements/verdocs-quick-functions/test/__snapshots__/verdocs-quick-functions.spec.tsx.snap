// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-quick-functions renders 1`] = `
<verdocs-quick-functions>
  <div class="container">
    <p class="title">
      Quick Create
    </p>
    <button>
      <svg enable-background="new 0 0 24 24" fill="#000000" height="24px" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
        <g>
          <rect fill="none" height="24" width="24"></rect>
          <g>
            <path d="M19,5v14H5V5H19 M19,3H5C3.9,3,3,3.9,3,5v14c0,1.1,0.9,2,2,2h14c1.1,0,2-0.9,2-2V5C21,3.9,20.1,3,19,3L19,3z"></path>
          </g>
          <path d="M14,17H7v-2h7V17z M17,13H7v-2h10V13z M17,9H7V7h10V9z"></path>
        </g>
      </svg>
      Template
    </button>
    <button>
      <svg fill="#000000" height="24px" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
        <path d="M0 0h24v24H0V0z" fill="none"></path>
        <path d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11z"></path>
      </svg>
      Document
    </button>
  </div>
</verdocs-quick-functions>
`;
